packages/components/src/number-control/index.tsx

```diff
@@ -64,16 +65,22 @@ function UnforwardedNumberControl(
 	const mergedRef = useMergeRefs( [ inputRef, forwardedRef ] );
 
 	const isStepAny = step === 'any';
-	const baseStep = isStepAny ? 1 : ensureNumber( step );
+	let baseStep = ensureNumber( step );
+	let baseSpin = ensureNumber( spinFactor ) * baseStep;
+	if ( isStepAny ) {
+		baseStep = 1;
+		baseSpin = 1 * ensureNumber( spinFactor );
+	}
 	const baseValue = roundClamp( 0, min, max, baseStep );
 	const constrainValue = (
 		value: number | string,
 		stepOverride?: number
-	) => {
+	): string => {
```

Comment:
```suggestion
	) => {
```

Nit: I think we can let TypeScript do the inference here.

Reply:
Oh yes, looks like we can! I thought this was helping solve the need for [these](https://github.com/WordPress/gutenberg/pull/52902/files#diff-1795d70ae6f25ef5ee79fa75d215a18e87de7d7bdc577b294376838325d983a8L123) @ts-expect-error comments, but it doesn't look like we need to specify the return type at all.